owncloudApp/src/main/java/com/owncloud/android/presentation/sharing/shares/PublicShareDialogFragment.kt

```diff
@@ -321,6 +343,111 @@ class PublicShareDialogFragment : DialogFragment() {
         })
     }
 
+    private fun requirementsPasswordPolicy(password: String, passwordPolicy: OCCapability.PasswordPolicy) {
+
+            var hasMaxCharacters = password.length <= passwordPolicy.maxCharacters!!
+            var hasMinCharacters = password.length >= passwordPolicy.minCharacters!!
+            var hasUpperCase = password.count { it.isUpperCase() } >= passwordPolicy.minUppercaseCharacters!!
+            var hasLowerCase = password.count { it.isLowerCase() } >= passwordPolicy.minLowercaseCharacters!!
+            var hasSpecialCharacter = password.count { SPECIALS_CHARACTERS.contains(it) } >= passwordPolicy.minSpecialCharacters!!
+            var hasDigit = password.count { it.isDigit() } >= passwordPolicy.minDigits!!
+
+            if (hasMaxCharacters) requirementChecked(binding.shareViaLinkPasswordPolicyMaxCharacters)
+            else requirementWarning(binding.shareViaLinkPasswordPolicyMaxCharacters)
+
+            if (hasMinCharacters) requirementChecked(binding.shareViaLinkPasswordPolicyMinCharacters)
+            else requirementWarning(binding.shareViaLinkPasswordPolicyMinCharacters)
+
+            if (hasUpperCase) requirementChecked(binding.shareViaLinkPasswordPolicyUpperCaseCharacters)
+            else requirementWarning(binding.shareViaLinkPasswordPolicyUpperCaseCharacters)
+
+            if (hasLowerCase) requirementChecked(binding.shareViaLinkPasswordPolicyLowerCaseCharacters)
+            else requirementWarning(binding.shareViaLinkPasswordPolicyLowerCaseCharacters)
+
+            if (hasSpecialCharacter) {
+                binding.shareViaLinkPasswordPolicyMinSpecialCharactersText.setTextColor(ContextCompat.getColor(appContext, R.color.success))
+                requirementChecked(binding.shareViaLinkPasswordPolicyMinSpecialCharactersIcon)
+            } else {
+                binding.shareViaLinkPasswordPolicyMinSpecialCharactersText.setTextColor(ContextCompat.getColor(appContext, R.color.warning))
+                requirementWarning(binding.shareViaLinkPasswordPolicyMinSpecialCharactersIcon)
+            }
+
+            if (hasDigit) requirementChecked(binding.shareViaLinkPasswordPolicyMinDigitCharacters)
+            else requirementWarning(binding.shareViaLinkPasswordPolicyMinDigitCharacters)
+
+
+            binding.shareViaLinkPasswordPolicyIntro.isVisible = true
+
+            if (passwordPolicy.minCharacters!! > 0) {
+                binding.shareViaLinkPasswordPolicyMinCharacters.text = getString(
+                    R.string.password_policy_min_characters, passwordPolicy.minCharacters
+                )
+                binding.shareViaLinkPasswordPolicyMinCharacters.isVisible = true
+            } else {
+                hasMinCharacters = true
+            }
+
```

Comment:
I think you managed this in a complex way and repeatedly. Why not doing everything related to a policy requirement (check and visibility) at the same time instead of checking all the policies and then handling all policies visibility, having to access all the capabilities values again? That way you can also add nullability check with a `let` and do everything related to that policy inside the block, and it would be much better structured

Reply:
Done, logic improved.